### PR TITLE
used log4j logging framework

### PR DESCRIPTION
**Pull request by erguncankecelioglu**
closes issue #4

### CODE REVIEW

File: src/main/java/tr/com/orioninc/laborant/controller/LabController.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ public String runCommand(Model model, @PathVariable String labName,
             model.addAttribute("currentLab", currentLab);
             try {
                 String commandResponse = labService.runCommandOnSelectedLab(labName, currentCommand.command);
-                System.out.println("INSIDE CONTROLLER" + commandResponse);
+                log.info("INSIDE CONTROLLER" + commandResponse);
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/controller/LabController.java]: Use parameterized logging instead of String concatenation

**Review by kemalyildirim — CHANGES_REQUESTED** [review]
Update all logging statements so that all will print meaningful logs.

**An example:**
`log.info(myarray);`
**A better example:**
`log.info("[startAppFunction] subscriber count is {}", myarray.size());`

**Comment by erguncankecelioglu** [comment]
- [x] I reviewed pull request myself

File: src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
+import tr.com.orioninc.laborant.controller.AdminController;
+import tr.com.orioninc.laborant.service.AdminService;
 
 @SpringBootApplication
+@Log4j2
 public class InternProjectApplication {
 
 	public static void main(String[] args) throws Exception
 	{
+		log.info(" [MAIN] Starting application");
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java]: * extra space before [MAIN]
* Please write the function name case sensetive, i.g. [main]

File: src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
 		SpringApplication.run(InternProjectApplication.class, args);
+
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java:14]: No change at all but shows as a diff, keep in mind in the future PR's please.

File: src/main/java/tr/com/orioninc/laborant/controller/LabController.java [hunk 4]
```diff
@@ -70,8 +71,9 @@ public String runCommandPage(Model model,@PathVariable String labName,@PathVaria
     public String runCommand(Model model, @PathVariable String labName,
                                  @ModelAttribute("currentCommand") CommandDTO currentCommand)
     {
+        log.info("[runCommand] Running command on lab named" + labName);
         if (currentCommand.getCommand() == "") {
-            System.out.println("EMPTY COMMAND");
+            log.warn("[EMPTY COMMAND]");
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/controller/LabController.java]:
```suggestion
            log.warn("[runCommand] Empty command.");
```

File: src/main/java/tr/com/orioninc/laborant/controller/LabController.java [hunk 5]
```diff
@@ -92,7 +94,7 @@ public String runCommand(Model model, @PathVariable String labName,
             model.addAttribute("currentLab", currentLab);
             try {
                 String commandResponse = labService.runCommandOnSelectedLab(labName, currentCommand.command);
-                System.out.println("INSIDE CONTROLLER" + commandResponse);
+                log.info("[INSIDE CONTROLLER]" + commandResponse);
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/controller/LabController.java]: Ditto

File: src/main/java/tr/com/orioninc/laborant/service/AdminService.java [hunk 6]
```diff
@@ -24,14 +26,18 @@ public  String addNewLab(String labName, String userName, String password, Strin
             if (Objects.isNull(searchHostUserPair)){
                 Lab labToBeAdded = new Lab(labName,userName,password,host,port);
                 labToBeAdded = labRepo.save(labToBeAdded);
+                log.info("[addNewLab] Successfully added new lab with name: " + labToBeAdded.getLabName());
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/AdminService.java]:
```suggestion
                log.info("[addNewLab] Successfully added new lab with name: {}", labToBeAdded.getLabName());
```

File: src/main/java/tr/com/orioninc/laborant/service/AdminService.java [hunk 7]
```diff
@@ -24,14 +26,18 @@ public  String addNewLab(String labName, String userName, String password, Strin
...
                 return "Successfully added the new lab named "+  labToBeAdded.getLabName() + " to the database";
             }
             else{
+                log.warn("[addNewLab] The lab named " + labName + " already exists in the database");
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/AdminService.java]: Ditto

File: src/main/java/tr/com/orioninc/laborant/service/AdminService.java [hunk 8]
```diff
@@ -24,14 +26,18 @@ public  String addNewLab(String labName, String userName, String password, Strin
...
                 return "There is already a pair in the database with username: "+userName + " and host: " + host ;
             }
         }
         else
         {
+
+            log.warn("[addNewLab] There is already a lab named " + labName + "  in the database. Try again");
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/AdminService.java]: Ditto

File: src/main/java/tr/com/orioninc/laborant/service/AdminService.java [hunk 9]
```diff
@@ -40,11 +46,12 @@ public  String addNewLab(String labName, String userName, String password, Strin
 
     public List<Lab> getAllLabs(){
         List<Lab> allLabs = labRepo.findAll();
+        log.info("[getAllLabs] - Getting all labs from the database... " + allLabs);
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/AdminService.java]: Ditto

File: src/main/java/tr/com/orioninc/laborant/service/AdminService.java [hunk 10]
```diff
@@ -40,11 +46,12 @@ public  String addNewLab(String labName, String userName, String password, Strin
...
         return allLabs;
-
     }
 
     public String deleteLab(String labName){
+        log.info("[deleteLab] lab deleted: " + labName);
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/AdminService.java]: Ditto

File: src/main/java/tr/com/orioninc/laborant/service/AdminService.java [hunk 11]
```diff
@@ -58,6 +65,7 @@ public String deleteLab(String labName){
     }
 
     public Lab findLabByName(String labName){
+        log.info("[findLabByName] Lab found: "+labName);
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/AdminService.java]: Ditto

File: src/main/java/tr/com/orioninc/laborant/service/LabService.java [hunk 12]
```diff
@@ -65,11 +67,9 @@ public String getALlLabsStatus(){
...
         else{
             String response = "";
-            String outputString = "";
-            for (Lab currentLab: allLabs
-                 )
-            {
-                outputString+= currentLab.getLabName() +   "    "+ "Host: "+ currentLab.getHost()  +  "      " ;
+            StringBuilder outputString = new StringBuilder();
+            for (Lab currentLab: allLabs) {
+                outputString.append("Lab: ").append(currentLab.getLabName()).append("  ").append("Host: ").append(currentLab.getHost()).append("----");
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/LabService.java]: Override toString method for Lab class and simply use `currentLab.toString()`

File: src/main/java/tr/com/orioninc/laborant/service/LabService.java [hunk 13]
```diff
@@ -87,17 +87,17 @@ public String getALlLabsStatus(){
                             words.add(tokenizer.nextToken());
                         outputArray.add(words);
 
-                        System.out.println(currentLine);
+                        log.info(currentLine); // I dont know what this info prints.
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/LabService.java]: Please debug the program and resolve what this log prints. Depending on information you can improve or remove the log line.

File: src/main/java/tr/com/orioninc/laborant/service/LabService.java [hunk 14]
```diff
@@ -87,17 +87,17 @@ public String getALlLabsStatus(){
...
                     }
-                    System.out.println(outputArray);
+                    log.info(outputArray); // I dont know what this info prints.
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/LabService.java]: Please debug the program and resolve what this log prints. Depending on information you can improve or remove the log line.

File: src/main/java/tr/com/orioninc/laborant/service/LabService.java [hunk 15]
```diff
@@ -106,27 +106,27 @@ public String getALlLabsStatus(){
...
-                    outputString+=  " \n";
-                    outputString+=response;
-                    outputString+="\n  \n";
+                    outputString.append(" \n");
+                    outputString.append(response);
+                    outputString.append("\n  \n");
                 }
                 catch(Exception e)
                 {
                     e.printStackTrace();
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/LabService.java:128]: Print this stack trace with logger in ERROR level:
`log.error("[getALlLabsStatus] exception while bla bla", e);`

File: src/main/java/tr/com/orioninc/laborant/service/LabService.java [hunk 16]
```diff
@@ -106,27 +106,27 @@ public String getALlLabsStatus(){
...
                 }
 
             }
-            System.out.println(outputString);
-            return outputString;
+            log.info("[LABSERVICE] " + outputString);
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/LabService.java]: Ditto

File: src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java [hunk 17]
```diff
@@ -1,14 +1,20 @@
 package tr.com.orioninc.laborant;
 
+import lombok.extern.log4j.Log4j2;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import tr.com.orioninc.laborant.controller.AdminController;
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java]: Unused import statement

File: src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java [hunk 18]
```diff
@@ -1,14 +1,20 @@
...
+import tr.com.orioninc.laborant.service.AdminService;
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java]: DItto

File: src/main/java/tr/com/orioninc/laborant/service/LabService.java [hunk 19]
```diff
@@ -44,32 +49,34 @@ public static String connectAndExecuteCommand(String username, String password,
         }
         catch(Exception e)
         {
+            log.error("[connectAndExecuteCommand] Error occured while connecting to lab: {}" , e.getMessage());
             e.printStackTrace();
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/LabService.java:56]: Please log the stack trace as well.
as I remember log4j2 loggers can take exceptions as a parameter

File: src/main/java/tr/com/orioninc/laborant/service/LabService.java [hunk 20]
```diff
@@ -44,32 +49,34 @@ public static String connectAndExecuteCommand(String username, String password,
...
                 channel.disconnect();
             }
-            return responseString;
         }
-
+        log.info("[connectAndExecuteCommand] Response string: {}", responseString);
+        return responseString; // it was inside finally block
     }
 
     public String getALlLabsStatus(){
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/LabService.java]: Function name is not following the naming convention
```suggestion
    public String getAllLabsStatus(){
```

File: src/main/java/tr/com/orioninc/laborant/service/LabService.java (hunk 15 above)
Comment (kemalyildirim) [reply src/main/java/tr/com/orioninc/laborant/service/LabService.java:128]: Ditto

File: src/main/java/tr/com/orioninc/laborant/service/LabService.java [hunk 21]
```diff
@@ -188,8 +199,10 @@ public String runCommandOnSelectedLab(String labName,String commandToBeExecuted)
             }
             catch(Exception e)
             {
+                log.error("[runCommandOnSelectedLab] error while running command on lab: {}",labName, e);
                 e.printStackTrace();
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/LabService.java:192]: Ditto

File: src/main/java/tr/com/orioninc/laborant/controller/LabController.java [hunk 22]
```diff
@@ -10,12 +11,9 @@
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Scanner;
-import java.util.StringTokenizer;
-
+import java.util.*;
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/controller/LabController.java]: Please avoid wildcard import statements, unless you're actually using all classes inside java.util

File: src/main/java/tr/com/orioninc/laborant/service/AdminService.java [hunk 23]
```diff
@@ -40,24 +46,26 @@ public  String addNewLab(String labName, String userName, String password, Strin
 
     public List<Lab> getAllLabs(){
         List<Lab> allLabs = labRepo.findAll();
+        log.info("[getAllLabs] - Getting all labs from the database... \n {}", allLabs);
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/AdminService.java]: Printing all labs is redundant. We already return the output.

File: src/main/java/tr/com/orioninc/laborant/service/AdminService.java [hunk 24]
```diff
@@ -40,24 +46,26 @@ public  String addNewLab(String labName, String userName, String password, Strin
...
             labRepo.delete(labToBeDeleted);
-
+            log.info("[deleteLab] Successfully deleted lab named {}", labName);
             return "Lab named "+labName+" is successfully deleted from the database";
 
         }
     }
 
     public Lab findLabByName(String labName){
+        log.info("[findLabByName] Lab found: {}",labName);
```
Comment (kemalyildirim) [inline src/main/java/tr/com/orioninc/laborant/service/AdminService.java]: Lab is not found yet. We can't be sure `labRepo.findByLabName(labName);` always runs successfully.

File: src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java (hunk 2 above)
Comment (erguncankecelioglu) [reply src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java]: done

File: src/main/java/tr/com/orioninc/laborant/controller/LabController.java (hunk 4 above)
Comment (erguncankecelioglu) [reply src/main/java/tr/com/orioninc/laborant/controller/LabController.java]: done

File: src/main/java/tr/com/orioninc/laborant/controller/LabController.java (hunk 5 above)
Comment (erguncankecelioglu) [reply src/main/java/tr/com/orioninc/laborant/controller/LabController.java]: done

File: src/main/java/tr/com/orioninc/laborant/service/AdminService.java (hunk 6 above)
Comment (erguncankecelioglu) [reply src/main/java/tr/com/orioninc/laborant/service/AdminService.java]: done

File: src/main/java/tr/com/orioninc/laborant/service/LabService.java (hunk 12 above)
Comment (erguncankecelioglu) [reply src/main/java/tr/com/orioninc/laborant/service/LabService.java]: done

File: src/main/java/tr/com/orioninc/laborant/service/LabService.java (hunk 16 above)
Comment (erguncankecelioglu) [reply src/main/java/tr/com/orioninc/laborant/service/LabService.java]: done

File: src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java (hunk 17 above)
Comment (erguncankecelioglu) [reply src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java]: optimized, done

File: src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java (hunk 18 above)
Comment (erguncankecelioglu) [reply src/main/java/tr/com/orioninc/laborant/InternProjectApplication.java]: done

File: src/main/java/tr/com/orioninc/laborant/service/LabService.java (hunk 19 above)
Comment (erguncankecelioglu) [reply src/main/java/tr/com/orioninc/laborant/service/LabService.java:56]: To be checked LabService.java line57

File: src/main/java/tr/com/orioninc/laborant/service/LabService.java (hunk 20 above)
Comment (erguncankecelioglu) [reply src/main/java/tr/com/orioninc/laborant/service/LabService.java]: refactored

File: src/main/java/tr/com/orioninc/laborant/controller/LabController.java (hunk 22 above)
Comment (erguncankecelioglu) [reply src/main/java/tr/com/orioninc/laborant/controller/LabController.java]: done

File: src/main/java/tr/com/orioninc/laborant/service/AdminService.java (hunk 23 above)
Comment (erguncankecelioglu) [reply src/main/java/tr/com/orioninc/laborant/service/AdminService.java]: done

File: src/main/java/tr/com/orioninc/laborant/service/AdminService.java (hunk 24 above)
Comment (erguncankecelioglu) [reply src/main/java/tr/com/orioninc/laborant/service/AdminService.java]: did changes on it

**Pull request closed without merging**